src/test/groovy/graphql/introspection/IntrospectionTest.groovy

```diff
@@ -435,4 +435,59 @@ class IntrospectionTest extends Specification {
         graphql.execute(query).data ==
                 [__type: [fields: [[args: [[defaultValue: '{inputField : "foo"}']]]]]]
     }
+
+    def "test parameterized introspection queries"() {
+        def spec = '''
+            scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")
+
+            directive @repeatableDirective(arg: String) repeatable on FIELD
+
+            """schema description"""
+            schema {
+                query: Query
+            }
+
+            directive @someDirective(
+                deprecatedArg : String @deprecated
+                notDeprecatedArg : String
+            ) repeatable on FIELD 
+
+            type Query {
+                """notDeprecated root field description"""
+               notDeprecated(arg : InputType @deprecated,  notDeprecatedArg : InputType) : Enum
+               tenDimensionalList : [[[[[[[[[[String]]]]]]]]]]
+            }
+            enum Enum {
+                RED @deprecated
+                BLUE
+            }
+            input InputType {
+                inputField : String @deprecated
+            }
+        '''
+
+        def graphQL = TestUtil.graphQL(spec).build()
+
+        def parseExecutionResult = {
+            [
+                it.data["__schema"]["types"].find{it["name"] == "Query"}["fields"].find{it["name"] == "notDeprecated"}["description"] != null, // descriptions is true
+                it.data["__schema"]["types"].find{it["name"] == "UUID"}["specifiedByURL"] != null, // specifiedByUrl is true
+                it.data["__schema"]["directives"].find{it["name"] == "repeatableDirective"}["isRepeatable"] != null, // directiveIsRepeatable is true
+                it.data["__schema"]["description"] != null, // schemaDescription is true
+                it.data["__schema"]["types"].find { it['name'] == 'InputType' }["inputFields"].find({ it["name"] == "inputField" }) != null // inputValueDeprecation is true
+            ]
+        }
+
+        when:
+            def allFalseExecutionResult = graphQL.execute(IntrospectionQuery.getIntrospectionQuery(false, false, false, false, false, 5))
+        then:
+            !parseExecutionResult(allFalseExecutionResult).any()
+            allFalseExecutionResult.data["__schema"]["types"].find{it["name"] == "Query"}["fields"].find{it["name"] == "tenDimensionalList"}["type"]["ofType"]["ofType"]["ofType"]["ofType"]["ofType"]["ofType"] == null // typeRefFragmentDepth is 5
+
+        when:
+            def allTrueExecutionResult = graphQL.execute(IntrospectionQuery.getIntrospectionQuery(true, true, true, true, true, 7))
+        then:
+            parseExecutionResult(allTrueExecutionResult).every()
+            allTrueExecutionResult.data["__schema"]["types"].find{it["name"] == "Query"}["fields"].find{it["name"] == "tenDimensionalList"}["type"]["ofType"]["ofType"]["ofType"]["ofType"]["ofType"]["ofType"]["ofType"]["ofType"] == null // typeRefFragmentDepth is 7
+    }
```

Comment:
To be sure that `IntrospectionQuery. INTROSPECTION_QUERY` actualyl equals what it used to eqal can we hard code a test here

It would copy the OLD string value here and then check that the new generated constant is the same

Reply:
There are differences/improvements in the spacing and formatting of the string produced by the AST printer. Perhaps  we can trim all whitespace and compare, or parse the old query with SchemaParser and reprint it?

Reply:
I've verified that the queries are equivalent via diff and added a test that ignores whitespace.